Sources/Transport/Streams/ProgramStream/ProgramStream.swift

```diff
@@ -20,13 +22,27 @@ extension ProgramStream {
 }
 
 public enum SecurityLayer {
-    case none, tls
+    case none
+    case tls(TLSConfig)
+}
+
+extension SecurityLayer: Equatable {
+    public static func ==(lhs: SecurityLayer, rhs: SecurityLayer) -> Bool {
+        switch (lhs, rhs) {
+        case (.none, .none):
+            return true
+        case (.tls(_), .tls(_)):
+            return true
```

Comment:
That's a good question - SecurityLayer, as an enum, mostly communicates the type of the security standard. I don't think the Configs should be compared, as we want `.tls(a) == .tls(b)` to be equal whatever a,b, I think. That's the tricky part with comparing enums with associated values.